### PR TITLE
add Query.getSingleResultOrNull()

### PR DESCRIPTION
**Pull request by gavinking**
Adds the new method I proposed in this discussion: https://github.com/jakartaee/persistence/issues/298#issuecomment-736682475.

### CODE REVIEW

**Comment by beikov** [comment]
Shouldn't it rather be something like this?

```java
    default T getSingleResultOrNull() {
        try {
            return getSingleResult();
        } catch (NoResultException e) {
            return null;
        }
    }
```

Given that `getResultStream` was also added with a default implementation.

**Comment by gavinking** [comment]
> Shouldn't it rather be something like this?

Sure, we could do that.

**Comment by mkalkowski83** [comment]
:+1:

**Review by lukasj — APPROVED** [review]

File: spec/src/main/asciidoc/ch03-entity-operations.adoc [hunk 1]
```diff
@@ -6082,19 +6145,18 @@ been joined to it, will be marked for rollback.
 
 *NoResultException*
 
-The _NoResultException_ is thrown by the
-persistence provider when _Query.getSingleResult_ or
-_TypedQuery.getSingleResult_ is invoked and there is no result to
-return. This exception will not cause the current transaction, if one is
+The _NoResultException_ is thrown by the persistence provider when
+_Query.getSingleResult_is invoked and there is no result to return.
```
Comment (gavinking) [inline spec/src/main/asciidoc/ch03-entity-operations.adoc]: oops!

**Comment by gavinking** [comment]
> add to changelog

Done.

This is ready to merge.

**Review by lukasj — APPROVED** [review]

**Review by lukasj — APPROVED** [review]

**Pull request merged**